kafka_enforcer_common/src/main/java/com/tesla/data/enforcer/EnforceCommand.java

```diff
@@ -79,13 +85,35 @@ protected EnforceCommand() {
         "for a test.");
   }
 
+  /**
+   * Initialize the enforcer with a zookeeper based admin client. Base classes must override this method.
```

Comment:
I wouldn't expect all Enforcer impls to use ZK, so wouldn't expect this requirement.

In general, it would be nice if we could find a way to instead have the enforcer impl "pick" the kind of client it needs (could easily see one that needs both Kafka and ZK) and give it access to standard providers for the admin client it needs

Reply:
addressed this by using ServiceLoaders: https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html

Enforcer implementations can now pick which of the available client providers to use, and can load the client directly from the provider